src/utils.rs

```diff
@@ -210,6 +246,44 @@ mod tests {
             load_wallet(&wallet_path).unwrap();
         })
     }
+
+    #[test]
+    fn test_should_replace_wallet() {
+        with_tmp_dir(|tmp_dir| {
+            let wallet_path = tmp_dir.join("wallet.json");
+            // simulate input
+            let input_nop = b"\n";
+            let input_yes = b"y\n";
+            let input_no = b"n\n";
+            // case: wallet path not exist
+            assert_eq!(
```

Comment:
Thanks for adding these tests!

nit: best practice is to split each test case into its own test function so that if one fails, it's easy to tell which one.

The test setup can be shared by putting the tests inside a module like: 

```
#[cfg(test)]
mod tests {
    use super::*;

    const INPUT_NOP: &[u8; 1] = b"\n";
    // other consts

    fn get_wallet_path() {
        tmp_dir.join("wallet.json")
    }

    fn remove_wallet() { 
       //...
    }

    #[test]
    fn test_should_replace_wallet_no_wallet() {
        remove_wallet();
        should_replace_wallet(&wallet_path, false, &input_nop[..]);
    }

    #[test]
    #[should_panic]
    fn test_should_replace_wallet_throws_err() {
        should_replace_wallet(&wallet_path, false, &input_nop[..]);
    }

   // other tests
}
```